libpod/container_validate.go

```diff
@@ -85,6 +85,15 @@ func (c *Container) validate() error {
 		return fmt.Errorf("cannot set static IP or MAC address if not creating a network namespace: %w", define.ErrInvalidArg)
 	}
 
+	// issue #25927 network-aliases cannot have spaces
+	for _, net := range c.config.Networks {
+		for _, alias := range net.Aliases {
+			if strings.Contains(alias, " ") {
```

Comment:
Maybe use ContainsAny and check for other invalid characters in DNS names - stuff like `&@*()!_#$`?

Reply:
(I wonder if there's a validation regex that does this?)

Reply:
regex is often slower and needs more meory/cpu as the regex must be compiled first.